python/gstaichi/lang/ast/ast_transformers/function_def_transformer.py

```diff
@@ -274,15 +334,23 @@ def _transform_func_arg(
 
     @staticmethod
     def _transform_as_func(ctx: ASTTransformerContext, node: ast.FunctionDef, args: ast.arguments) -> None:
+        assert ctx.argument_data is not None
+        assert ctx.func is not None
         for data_i, data in enumerate(ctx.argument_data):
-            argument = ctx.func.arguments[data_i]
+            argument = ctx.func.arg_metas[data_i]
             FunctionDefTransformer._transform_func_arg(
                 ctx,
                 argument.name,
                 argument.annotation,
                 data,
             )
 
+        # deal with dataclasses
+
+        # pylint: disable=import-outside-toplevel
+        from gstaichi.lang.kernel_impl import Func
```

Comment:
If you need delayed import, I would recommend to rather move import on top of this method but as you wish.